### PR TITLE
Convert ExDTLS into module

### PR DESCRIPTION
**Pull request by mickel8**

### CODE REVIEW

File: c_src/ex_dtls/native.spec.exs [hunk 1]
```diff
@@ -4,29 +4,28 @@ interface NIF
...
-spec process(state, packets :: payload) :: {:ok :: label, state, packets :: payload}
-                                           | {(:hsk_want_read :: label), state}
-                                           | {{:hsk_packets :: label, packets :: payload}, state}
-                                           | {{:hsk_finished :: label,
-                                              client_keying_material :: payload,
-                                              server_keying_material :: payload,
-                                              protection_profile :: int,
-                                              packets :: payload}, state}
-                                           | {{:connection_closed :: label, :peer_closed_for_writing :: label}, state}
+spec process(state, packets :: payload) ::
```
Comment (mickel8) [inline c_src/ex_dtls/native.spec.exs:25]: I feel like `receive` would fit here better but it overlaps with Elixir. There also are `recv`, `recv_data`, `feed`, `read`

Comment (LVala) [reply c_src/ex_dtls/native.spec.exs:25]: `handle_data`? We already have `handle_timeout`, but no strong opinions from me.

**Review by LVala — APPROVED** [review]
I'm don't know much about NIFs, but overall SGTM, except for 1 pet peeve - you cannot create the certs without doing `ExDTLS.init/1` which is annoying when the DTLS role is not known beforehand. Are you planning to do something about that? Also the fingerprint validation stuff, but that's also for another PR.

**Comment by mickel8** [comment]
> you cannot create the certs without doing ExDTLS.init/1 which is annoying when the DTLS role is not known beforehand

Fixed 

> Are you planning to do something about that? Also the fingerprint validation stuff, but that's also for another PR.

Yeap, probably in the next PR

**Comment by mat-hek** [comment]
BTW, are you going to update the membrane_ice_plugin to use the new ex_dlts API?

**Pull request merged**